raiden-ts/src/raiden.ts

```diff
@@ -1471,20 +1461,29 @@ export class Raiden {
             mergeMap(async (response) => response.text()),
             map((text) => {
               const suggestions = decode(SuggestedPartners, jsonParse(text));
-              assert(suggestions.length, ['empty partner suggestion from PFS', { pfs: pfs.url }]);
+
+              if (!suggestions.length) {
+                firstResponse = [];
+              }
```

Comment:
Are you sure this works? If we get an empty array back and we do not manually throw (as we did with the assert before) this will just return the empty array. This means the `first()` operator just goes straight ahead and does not query any further PFS. And I think that's what we want. All you logic about the `firstResponse` can be also an empty array will never be used.
I think you should extend the test cases. So you want to have multiple PFS. The first returns an empty list. Then one case where all following PFS throw, so you expect to get the empty list of the first. And one test case where any future PFS in the list return a positive array, so you expect that one to get returned in the end.
The CodeCov remark on this PR is a hint that you either lack in tests cases or - what I stated - your code in line 1481 never gets applied.

Sorry if I'm totally wrong. :see_no_evil:

Reply:
Eventually this is easily solvable by adding a filter like `first(suggestions => first.length)`. Not too sure about the rest as we still have no error. xD

Reply:
No, you're right. For how I intended it to work there's at least a early return missing, but there might be additional problems., I'll write more tests.

Reply:
As @weilbith stated, you should set `firstResponse` to the empty array **and** throw on the map, to get the next pfs in queue to be tried. An alternative would be to abuse the error handling in JS inside `map` to `throw suggestions`, which is the empty array (and not an error, as usually advised), so it can be both set to `firstResponse` and omitted on the `catchError`